internal/loginsrv/input.go

```diff
@@ -0,0 +1,58 @@
+package loginsrv
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// ReadLine waits for the manual login input to send the session token.
+// NOTE: As we are dealing with a Read, cancelling the context will not unblock.
+//       The caller is expected to close the reader.
+func ReadLine(ctx context.Context, r io.Reader, w io.Writer, tokenChan chan<- string) error {
+	// Wrap the reader with bufio to simplify the readline.
+	buf := bufio.NewReader(r)
+
+retry:
+	_, _ = fmt.Fprintf(w, "or enter token manually: ") // Best effort. Can only fail on custom writers.
```

Comment:
This message doesn't seem descriptive enough... maybe we should include specific direction so the user knows to paste the contents of the URL bar.

Reply:
This is the same message that concourse ci uses, with the same formatting. We expect most people to not require this part, so I think it might be good enough? I am not really good with words ^^, If you have a suggestion for better wording, I'd be happy to update.